### PR TITLE
Refresh tests

### PR DESCRIPTION
**Pull request by mpharrigan**
- Switch to pytest (nosetests no longer maintained?)
 - use "fixtures" instead of `yield` (later deprecated by pytest)
 - minor improvements
 - start switching to relative imports
 - update travis to test on python 3.6 and numpy 1.12
 - remove test data from package
 - (related to above) move `get_fn` out of mdtraj proper
 - move geometry tests to be with the rest of the tests
 - port test_examples as well. TODO: fix some of these?
 - remove nose from conda recipe

The biggest changes are in test_trajectory.py. I changed it to programatically generate NxN tests for all formats. Sometimes these tests don't make sense due to the nature of the format

I've marked several things as xfail (expected fail). My intent is to open issues for "true" failures and eventually fix them. Things that aren't supported by the file formats themselves should be marked skip.

### CODE REVIEW

**Comment by mpharrigan** [comment]
Also start switching to relative imports

**Comment by mpharrigan** [comment]
The biggest changes are in `test_trajectory.py`. I changed it to programatically generate `NxN` tests for all formats. Sometimes these tests don't make sense due to the nature of the format:

I've marked  several things as xfail (expected fail) and I'm not quite done triaging all the failures in `test_trajectory.py`. My intent is to open issues for "true" failures and eventually fix them. Things that aren't supported by the file formats themselves should be marked `skip`.

**Comment by mpharrigan** [comment]
I've continued with this PR with the following additions

 - remove test data from package
 - (related to above) move `get_fn` out of mdtraj proper
 - move geometry tests to be with the rest of the tests

File: tests/test_nmr.py [hunk 1]
```diff
@@ -34,8 +35,8 @@ def test_ppm(get_fn):
     #       2     ASN   CA    999.000  53.004  51.168  51.802  53.081  54.098  52.820  52.379  51.856  53.034  52.754  54.134  54.222  51.210  52.207  50.824  54.459  53.605  54.211  53.688  52.344  53.004  51.168  51.802  53.081  54.098  52.820  52.379  51.856  53.034  52.754  54.134  54.222  51.210  52.207  50.824  54.459  53.605  54.211  53.688  52.344
 
 
-@pytest.mark.skipif(not (find_executable(SHIFTX2) and six.PY2),
-                    reason='SHIFTX2 binary not found or Python 2 not found.')
+@pytest.mark.skipif((not find_executable(SHIFTX2)) or (not six.PY2) or (os.environ.get('TRAVIS', '') == 'true'),
```
Comment (ctk3b) [inline tests/test_nmr.py]: 🎉

Comment (mpharrigan) [reply tests/test_nmr.py]: indeed :) test works fine locally!

**Pull request merged**